Core/src/org/sleuthkit/autopsy/communications/AccountDeviceInstanceNode.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2017-18 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.communications;
+
+import java.awt.event.ActionEvent;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import javax.swing.AbstractAction;
+import javax.swing.Action;
+import org.openide.nodes.AbstractNode;
+import org.openide.nodes.Children;
+import org.openide.nodes.Sheet;
+import org.openide.util.NbBundle;
+import org.openide.util.Utilities;
+import org.openide.util.lookup.Lookups;
+import org.sleuthkit.autopsy.datamodel.NodeProperty;
+import org.sleuthkit.datamodel.Account;
+import org.sleuthkit.datamodel.AccountDeviceInstance;
+import org.sleuthkit.datamodel.CommunicationsFilter;
+import org.sleuthkit.datamodel.CommunicationsManager;
+
+/**
+ * Node to represent an Account Device Instance in the CVT
+ */
+final class AccountDeviceInstanceNode extends AbstractNode {
+
+    private final AccountDeviceInstanceKey accountDeviceInstanceKey;
+    private final CommunicationsManager commsManager;
+    private final Account account;
+
+    AccountDeviceInstanceNode(AccountDeviceInstanceKey accountDeviceInstanceKey, CommunicationsManager commsManager) {
+        super(Children.LEAF, Lookups.fixed(accountDeviceInstanceKey, commsManager));
+        this.accountDeviceInstanceKey = accountDeviceInstanceKey;
+        this.commsManager = commsManager;
+        this.account = accountDeviceInstanceKey.getAccountDeviceInstance().getAccount();
+        setName(account.getTypeSpecificID());
+        setIconBaseWithExtension("org/sleuthkit/autopsy/communications/images/" + Utils.getIconFileName(account.getAccountType()));
+    }
+
+    public AccountDeviceInstance getAccountDeviceInstance() {
+        return accountDeviceInstanceKey.getAccountDeviceInstance();
+    }
+
+    public AccountDeviceInstanceKey getAccountDeviceInstanceKey() {
+        return accountDeviceInstanceKey;
+    }
```

Comment:
I would give the methods that are not overrides package access, since the class is package access.  That way, if we decide to go public, the developer who does this will need to think about what they want for the public API. This method, for example, seems to be something that would only need package access.

Also, I am curious about why a node would need to keep a reference to its key. Could this make key refernces hang around after they could otherwise be garbage collected, i.e., do we understand the way the keys for which we pass of ownership to the NetBeans framework should be managed in memory?

Reply:
I suppose the node does not need to keep a reference to its key.  I am no sure we have a complete understanding of how Netbeans manages keys in memory.  @esaunders  might have thoughts since he did most of the work to reduce memory consumption of retained keys and nodes.